pkg/tasks/tasks_test.go

```diff
@@ -134,3 +136,111 @@ func TestNewCmdTask(t *testing.T) {
 		t.Errorf("expected writer to receive the following content: \n%s\n. But instead it received: %s", expectedContent, actualContent)
 	}
 }
+
+// A dummy reader that simply yields as many blank lines as requested. The only
+// thing we want to do with the output is count the number of lines.
+type BlankLineReader struct {
+	totalLinesToYield int
+	linesYielded      int
+}
+
+func (d *BlankLineReader) Read(p []byte) (n int, err error) {
+	if d.totalLinesToYield == d.linesYielded {
+		return 0, io.EOF
+	}
+
+	d.linesYielded += 1
+	p[0] = '\n'
+	return 1, nil
+}
+
+func TestNewCmdTaskRefresh(t *testing.T) {
+	type scenario struct {
+		name                        string
+		totalTaskLines              int
+		linesToRead                 LinesToRead
+		expectedLineCountsOnRefresh []int
+	}
+
+	scenarios := []scenario{
+		{
+			"total < initialRefreshAfter",
+			150,
+			LinesToRead{100, 120},
+			[]int{100, 100},
```

Comment:
We get one extra refresh here (and for the other tests below), I suppose it's from after handling the stop command. Maybe the whole setup can be done better, feel free to improve!

Reply:
I've pushed a commit to fix that up :)

Reply:
Thanks! :)  Happy to keep it as a separate commit, it's nice to see how the tests change again with this change.

Ready to merge from my side.